src/gui/optionsdlg.cpp

```diff
@@ -445,25 +454,19 @@ void OptionsDialog::changePage(QListWidgetItem *current, QListWidgetItem *previo
 
 void OptionsDialog::loadWindowState()
 {
-    const Preferences* const pref = Preferences::instance();
-
-    resize(pref->getPrefSize(this->size()));
+    const QSize defaultSize = size() * Utils::Gui::screenScalingFactor(this);
```

Comment:
...or, at least, `QSize Utils::Gui::scaledSize(const QWidget *)`

Reply:
You don't want to have this one too?
`size * scalingFactor` is occurring very often (like resize).

Reply:
Or you can add the second optional param to  your resize util: `resize(QWidget *widget, const QSize &baseSize = QSize())`.

Reply:
added `scalingFactor()` and added second parameter to `resize()`.